pkg/dcl/clientconfig/config.go

```diff
@@ -40,15 +40,15 @@ type Options struct {
 	controller.Config
 }
 
-func New(ctx context.Context, opt Options) (*dcl.Config, error) {
+func NewWithHttpClient(ctx context.Context, opt Options) (*dcl.Config, *http.Client, error) {
```

Comment:
Nit: NewConfigAndClient (as you did below) is probably a clearer name. I would expect `...WithHttpClient` to take the http.Client as input.